File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    open = '(['
    close = ')]'

    stack = []
    for char in s:
        if char in open:
            stack.append(char)
        elif char in close:
            if not stack:
                return False
            if open.index(stack.pop()) != close.index(char):
                return False

    return len(stack) == 0

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.Stack;
class Solution {
    boolean solution(String s) {
        Stack<String> stack = new Stack<String>();
	    boolean answer = true;

        for(int i=0; i<s.length(); i++) {
            char c = s.charAt(i);
            if(c == '('){
                stack.push(s);
            }else if(c==')') {
                if(stack.isEmpty()) {
                    answer = false;
                    break;
                }
                stack.pop();
            }
        }
        if(!stack.isEmpty()) {
            answer = false;
        }


        return answer;
    }
}